src/packaging/dmlinux/kernel/Makefile

```diff
@@ -0,0 +1,20 @@
+VDO_VERSION = %%VDOVersion%%
+
+OBJECTS = $(patsubst %.c,dm-vdo/%.o,$(notdir $(wildcard $(src)/dm-vdo/*.c)))
+
+INCLUDES = -I$(src)/dm-vdo
+
+EXTRA_CFLAGS =	-std=gnu11					\
+		-fno-builtin-memset				\
+		-fno-omit-frame-pointer				\
+		-fno-optimize-sibling-calls			\
+		-Werror						\
+		$(if $(CONFIG_KASAN),,-Wframe-larger-than=400)	\
+		-DVDO_VERSION=\"$(VDO_VERSION)\"		\
+		-DSTATIC=					\
+		-DVDO_INTERNAL					\
+		$(INCLUDES)
```

Comment:
We shoudl reconsider these compile flags to more closely reflect what the kernel does. I suppose we can defer that step to a later change, though.

At a minimum, the STATIC and VDO_INTERNAL lines should not be needed because all that stuff should already have been replaced in the kernel sources. (And we don't want to mask stuff if we make a mistake.) The VDO_VERSION line is similarly unnecessary because the upstream version current hard-codes a version number.

Reply:
I don't think we want to mess with the version number for now. We use it in the other srpms/rpms to depend on a particular version or higher. If we try to plug in the kernel version, things are gonna break. I think we need to look at the versioning stuff as a whole later on.

Reply:
My first thought is that if we've got other rpms relying on this then we have a bigger problem to sort out. The vdo rpms already have to work with real kernels so there should be some way to address the issue. But we can defer it to a later change.